src/backend/optimizer/util/clauses.c

```diff
@@ -2631,19 +2631,23 @@ eval_const_expressions_mutator(Node *node,
 				List	   *args = expr->args;
 				Expr	   *simple;
 				OpExpr	   *newexpr;
+				int32       result_typmod = -1;
 
 				/*
 				 * Need to get OID of underlying function.  Okay to scribble
 				 * on input to this extent.
 				 */
 				set_opfuncid(expr);
 
+				if (sql_dialect == SQL_DIALECT_TSQL && pltsql_exprTypmod_hook)
```

Comment:
please explain why are we handling only handful of expr type? what about other node types?

Reply:
we need to set the correct typmod only for those nodes whose typmod is set as `-1`. For other nodes, the typmod is already set correctly.